apps/pigment-css-vite-app/src/main.tsx

```diff
@@ -1,13 +1,40 @@
 import * as ReactDOMClient from 'react-dom/client';
 import * as React from 'react';
-import { BrowserRouter as Router, useRoutes } from 'react-router-dom';
+import { BrowserRouter as Router, useLocation, useRoutes } from 'react-router-dom';
 import { ThemeProvider, createTheme } from '@mui/material/styles';
+import CircularProgress from '@mui/material/CircularProgress';
 import CssBaseline from '@mui/material/CssBaseline';
+import { css } from '@pigment-css/react';
+import { Box } from '@pigment-css/react/Box';
+import { ErrorBoundary } from 'react-error-boundary';
 import routes from '~react-pages';
 import '@pigment-css/react/styles.css';
+import { ErrorBoundaryFallback } from './components/ErrorBoundaryFallback';
 
 function App() {
-  return <React.Suspense fallback={<p>Loading...</p>}>{useRoutes(routes)}</React.Suspense>;
+  const location = useLocation();
+  return (
+    <ErrorBoundary key={location.pathname} FallbackComponent={ErrorBoundaryFallback}>
+      <React.Suspense
+        fallback={
+          <Box
+            as="div"
+            className={css`
+              width: 100vw;
+              height: 100vh;
+              display: flex;
+              justify-content: center;
+              align-items: center;
+            `}
```

Comment:
why not `sx`? if it's for testing, it's fine for me.

Reply:
Mainly because I was copy pasting and sx only supports object syntax.